### PR TITLE
feat(docs):  batch 4 of proofread/edited docs

### PR DESCRIPTION
**Pull request by vwheeler63**
- buttonmatrix.rst
- calendar.rst
- chart.rst
- checkbox.rst
- dropdown.rst

NOTE:  MAJOR work went into `checkbox.rst` (took a full day by itself).  Reason:

- it was completely missing introductory/orientation material, now provided;
- there was much missing detail and conflicting information that had to be sorted
  out by studying the source code;
- the missing introductory/orientation material had to be gathered by
  studying the source code;
- ambiguous/unclear content had to be sorted out by either studying the source code
  and/or setting up example code to test it.

This is presented here, and is a HUUUUUUGE improvement and speed-up for programmers
who have yet to learn how to use these Widgets.  Programmers learning to use Chart will
now not need to sort out the missing details by studying the source as I had to herein (and
if I were using it in a firmware project, I would have had to do the same).

cc:  @kisvegabor @liamHowatt @FASTSHIFT

(You guys are CC'd since you have been typically the one reviewing my submissions.  Thank you for all you've done so far!)

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.  Done.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.  n/a
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.  n/a
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).  n/a
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).  n/a  (No code was changed.)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.  Done.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.  Done.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by vwheeler63** [comment]
Note:  I removed `buttonmatrix.rst` because I realized it is already in PR #7194.

**Review by kisvegabor — DISMISSED** [review]
Look great, thank you! 

`shown_moth` for the calendar will be added a separate PR?

File: docs/details/widgets/checkbox.rst [hunk 1]
```diff
@@ -33,27 +35,27 @@ Usage
...
-The text can be modified with the
-:cpp:expr:`lv_checkbox_set_text(cb, "New text")` function and will be
+The text can be modified with
+:cpp:expr:`lv_checkbox_set_text(cb, "New text")` and will be
 dynamically allocated.
 
-To set a static text, use :cpp:expr:`lv_checkbox_set_static_text(cb, txt)`. This
-way, only a pointer to ``txt`` will be stored. The text then shouldn't
-be deallocated while the checkbox exists.
+To set static text, use :cpp:expr:`lv_checkbox_set_static_text(cb, txt)`. This
```
Comment (liamHowatt) [inline docs/details/widgets/checkbox.rst]:
```suggestion
To set static text, use :cpp:expr:`lv_checkbox_set_text_static(cb, txt)`. This
```

**Review by liamHowatt — COMMENTED** [review]
> `shown_moth` for the calendar will be added a separate PR?

Good catch. Looks like the API is now called `lv_calendar_set_showed_date` so the hyperlink isn't created.

Besides that, Calendar looks great!

**Comment by vwheeler63** [comment]
@kisvegabor @liamHowatt  Liam, good catch!  And while I was discussing the `lv_calendar_set_showed_date()` function-name change with Gábor, I forgot to rename it back to its CURRENT name in the `calendar.c`, so I broke that link as well.  Both are now fixed.  And you can see the fixed links here.

Then in the PR with the function name change, I will change it in all 3 places: `calendar.rst` as well as `calendar.c` and `calendar.h` so the hyperlinks remain intact.

The updated version is reflected here:

http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/details/widgets/calendar.html

**Review by liamHowatt — APPROVED** [review]
> so I thought I would separate code changes into a separate PR to make it easier on the reviewers.

Totally get it!

**Review by kisvegabor — APPROVED** [review]

File: docs/details/widgets/chart.rst [hunk 2]
```diff
@@ -4,110 +4,241 @@
...
+
+can be used as well.
+
+Modifying data
+--------------
+
+You have several options to set the Y-values for a data series:
+
+1. Set the values programmatically in the array like ``ser1->points[3] = 7`` and refresh the
+   chart with :cpp:enumerator:`lv_chart_refresh(chart)`.
```
Comment (liamHowatt) [inline docs/details/widgets/chart.rst:223]: I only just spotted this while using your changes as a reference for writing new docs :smile:

If you edit this file again in another PR and you remember this comment, please make this ":cpp:expr:\`lv_chart_refresh(chart)\`"

Comment (vwheeler63) [reply docs/details/widgets/chart.rst:223]: > I only just spotted this while using your changes as a reference for writing new docs 😄

Oh my goodness!  **Done.**  I have quite a few "minor" edits in docs already submitted that I am planning to submit as a "wrap-up" once I get through all the Widgets.

**Pull request merged**